### ISSUE TITLE
Bohrium Compilation on OS X

### ISSUE THREAD

**Issue by ZhijieWang** [bh107/bohrium#21]
Hi, I have OSX 10.9 and 

```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
```

Unfortunately, the simple brew install doesn't work, complaining SHA1 mismatch. So i did a manual compile.

However, during execution  got this.

```
AttributeError: 'module' object has no attribute 'genfromtxt'
!! Trying to bind but compiled without required library  !!
!!            Install hwloc and re-compile               !!
!!                          OR                           !!
!! Disable binding by setting ENV_VAR: BH_VE_CPU_BIND=0  !!
```

I got the no attibute part resolved by using original numpy to create the array and passing the array to bohrium to calculate.

I did have hwloc installed via homebrew.

And I did get an error at the end
[1]    30772 bus error  python markovchaingpu.py

**Comment by kenkendk** [comment]
The hwloc module is not used on OSX, see issue #17.

Could you post the full stack trace from the exception?
Also, I do not recognize the `markovchaingpu.py` filename, can you post the source? Or at least the code around the crash site?

**Comment by ZhijieWang** [comment]
The error message didn't tell me anything about the line. I will get a chance and put up the source code.

**Comment by ZhijieWang** [comment]
import numpy as numpy
import bohrium as np
import time
start_time = time.time()
n = 10
trades=[]
trades_raw = numpy.genfromtxt('fullinput.csv', delimiter=",")
for i in range(0,n) :
	for t in trades_raw:
		temp = np.zeros(18)
		temp[t[1]-1]=t[0]
		trades.append(temp)

trades = numpy.swapaxes(np.asarray(trades),0,1)
print trades.shape
mat_a = np.genfromtxt('US_2014_CI_Q4.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_b = np.genfromtxt('US_2015_CI_Q1.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_c = np.genfromtxt('US_2015_CI_Q2.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_d = np.genfromtxt('US_2015_CI_Q3.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_e = np.genfromtxt('US_2015_CI_Q4.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_f = np.genfromtxt('US_2016_CI_Q1.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_g = np.genfromtxt('US_2016_CI_Q2.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_h = np.genfromtxt('US_2016_CI_Q3.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
mat_i = np.genfromtxt('US_2016_CI_Q4.csv', delimiter=',').astype("float64")[:,2].reshape(18,18)
#print (mat_a.shape)
calculation_time = time.time()
trades = np.dot(mat_a, trades)
trades = np.dot(mat_b, trades)
trades = np.dot(mat_c, trades)
trades = np.dot(mat_d, trades)
trades = np.dot(mat_e, trades)
trades = np.dot(mat_f, trades)
trades = np.dot(mat_g, trades)
trades = np.dot(mat_h, trades)
trades = np.dot(mat_i, trades)

print("---total  %s seconds ---" % (time.time() - start_time))
print("---calculation %s seconds ---" % (time.time() - calculation_time))

np.savetxt("result.csv", trades, delimiter=",")

**Comment by kenkendk** [comment]
I can reproduce.

Using this code instead (I don't have the file):
```python
import numpy as numpy
import bohrium as np
n = 10
trades=[]
trades_raw = numpy.zeros((2,18))

for i in range(0,n) :
  for t in trades_raw:
    temp = np.zeros(18)
    temp[t[1]-1]=t[0]
    trades.append(temp)

#crash happens here
trades = numpy.swapaxes(np.asarray(trades),0,1)
print trades.shape
```


Stacktrace:
```
0   libsystem_platform.dylib      	0x00007fff887bb080 _platform_memmove$VARIANT$Unknown + 224
1   multiarray.so                 	0x0000000101cbd4b8 _contig_to_contig + 19
2   multiarray.so                 	0x0000000101c7c37f raw_array_assign_array + 667
3   multiarray.so                 	0x0000000101c7cc02 PyArray_AssignArray + 835
4   multiarray.so                 	0x0000000101c86292 setArrayFromSequence + 133
5   multiarray.so                 	0x0000000101c864e6 setArrayFromSequence + 729
6   multiarray.so                 	0x0000000101c88c42 PyArray_FromAny + 878
7   multiarray.so                 	0x0000000101c8901c PyArray_CheckFromAny + 257
8   multiarray.so                 	0x0000000101ce4b1c _array_fromobject + 746
9   org.python.python             	0x000000010194734d PyEval_EvalFrameEx + 17184
10  org.python.python             	0x0000000101942df9 PyEval_EvalCodeEx + 1412
11  org.python.python             	0x00000001018e731c function_call + 350
12  ndarray.so                    	0x00000001025d840f __pyx_pw_7ndarray_20fix_returned_biclass_1inner + 191 (ndarray.c:5771)
13  org.python.python             	0x00000001018c93b6 PyObject_Call + 99
14  org.python.python             	0x0000000101945e8c PyEval_EvalFrameEx + 11871
15  org.python.python             	0x0000000101942df9 PyEval_EvalCodeEx + 1412
16  org.python.python             	0x000000010194286f PyEval_EvalCode + 54
17  org.python.python             	0x0000000101962a2a run_mod + 53
18  org.python.python             	0x0000000101962acd PyRun_FileExFlags + 133
19  org.python.python             	0x000000010196266a PyRun_SimpleFileExFlags + 769
20  org.python.python             	0x0000000101973d6f Py_Main + 3051
21  libdyld.dylib                 	0x00007fff91c245c9 start + 1
```

**Comment by madsbk** [comment]
Fixed in commit efad230ab73eae745272f251f6861a17885362fb
